tests/drucker_prager_derivatives/screen-output

```diff
@@ -0,0 +1,220 @@
+onezero 2: Finite difference = -2.49047e+31. Analytical derivative = -2.49047e+31
+onezero 3: Finite difference = -9.86089e+36. Analytical derivative = -9.99827e+36
+   Error: The derivative of the viscosity to the strain rate is too different from the analitical value.
+onezero 4: Finite difference = -2.26757e+33. Analytical derivative = -2.26757e+33
+oneone 0: Finite difference = -5.94041e+30. Analytical derivative = -5.94041e+30
+oneone 1: Finite difference = -3.39214e+29. Analytical derivative = -3.39218e+29
+oneone 2: Finite difference = -1.36975e+30. Analytical derivative = -1.36976e+30
+oneone 3: Finite difference = 0. Analytical derivative = -0
+oneone 4: Finite difference = -5.03316e+25. Analytical derivative = -0
+   Error: The derivative of the viscosity to the strain rate is too different from the analitical value.
+Some parts of the test where not succesfull.
+
+Test for p = 1000
+pressure 0: Finite difference = 5.42254e+10. Analytical derivative = 5.42254e+10
+pressure 1: Finite difference = 4.2881e+10. Analytical derivative = 4.2881e+10
+pressure 2: Finite difference = 2.49368e+10. Analytical derivative = 2.49368e+10
+pressure 3: Finite difference = 496606. Analytical derivative = 489830
+   Error: The derivative of the viscosity to the pressure is too different from the analitical value.
+pressure 4: Finite difference = 2.26753e+10. Analytical derivative = 2.26753e+10
+zerozero 0: Finite difference = 5.94041e+30. Analytical derivative = 5.94041e+30
+zerozero 1: Finite difference = 3.39217e+29. Analytical derivative = 3.39218e+29
+zerozero 2: Finite difference = 1.36976e+30. Analytical derivative = 1.36976e+30
+zerozero 3: Finite difference = 0. Analytical derivative = -0
+zerozero 4: Finite difference = -5.03316e+25. Analytical derivative = -0
+   Error: The derivative of the viscosity to the strain rate is too different from the analitical value.
+onezero 0: Finite difference = -1.32009e+30. Analytical derivative = -1.32009e+30
+onezero 1: Finite difference = 1.85205e+31. Analytical derivative = 1.85205e+31
+onezero 2: Finite difference = -2.49047e+31. Analytical derivative = -2.49047e+31
+onezero 3: Finite difference = -9.86089e+36. Analytical derivative = -9.99827e+36
+   Error: The derivative of the viscosity to the strain rate is too different from the analitical value.
+onezero 4: Finite difference = -2.26757e+33. Analytical derivative = -2.26757e+33
+oneone 0: Finite difference = -5.94041e+30. Analytical derivative = -5.94041e+30
+oneone 1: Finite difference = -3.39214e+29. Analytical derivative = -3.39218e+29
+oneone 2: Finite difference = -1.36975e+30. Analytical derivative = -1.36976e+30
+oneone 3: Finite difference = 0. Analytical derivative = -0
+oneone 4: Finite difference = -5.03316e+25. Analytical derivative = -0
+   Error: The derivative of the viscosity to the strain rate is too different from the analitical value.
+Some parts of the test where not succesfull.
```

Comment:
ah, right. I had written a bit about that in the big pull request, but forgot to copy it to here to discuss. 

I had written something on this in a email to wolfgang a long time ago:

In short the problem is that for all common cases I found that the finite difference and the analytical solution are exactly the same. But with a strain rate in which the diagonal entries are the same the analytical solution gives -0 and the finite difference gives results on the order of the other strain rates (~-1e+20).

Here is the relevant code (There is also a minimum to edot_ii in the real code.):
```
   const double edot_ii = 2 * std::sqrt(0.5*deviator(in.strain_rate[i])*deviator(in.strain_rate[i]));
   const double stress_exponent_inv = (1./stress_exponent[c]);
   composition_viscosities[c] = std::pow(prefactor[c],-stress_exponent_inv) * std::pow(edot_ii,stress_exponent_inv-1);
   composition_viscosities_derivatives[c] = 2 * (stress_exponent_inv-1) * composition_viscosities[c] * (1/(edot_ii*edot_ii)) * deviator(in.strain_rate[i]);
```
I think this happens for the following reason: In the analytical equation, the deviator make the diagonal then zero, and the analytical solution then also ends up with zero entries on the diagonal. The finite difference method first gives the original strain rate tensor to the material model, and the deviator makes the diagonal zero. But then the finite difference method perturbs only one of the diagonal entries, which make the diagonal entries not equal any more and as a consequence the diagonal of the deviator will not contain zero entries any more, and I think this lead to a viscosity change on the order of what normally happens. 

Thoughts?